docs/federate.md

```diff
@@ -0,0 +1,156 @@
+Setting up Federation
+=====================
+
+Federation is the process by which users on different servers can participate
+in the same room. For this to work, those other servers must be able to contact
+yours to send messages.
+
+The ``server_name`` configured in the Synapse configuration file (often
+``homeserver.yaml``) defines how resources (users, rooms,...) will be
+identified (ie: ``@user:example.com``, ``#room:example.com``). By
+default, it is also the default domain that other servers will use to
+try to reach your server via port 8448. This is easy to set
+up and will work provided you set the ``server_name`` to match your
+machine's public DNS hostname, and provide Synapse with a TLS certificate
+which is valid for your ``server_name``.
+
+For a more flexible configuration, you can have ``server_name``
+resources (ie: ``@user:example.com``) served by a different host and
+port (ie: ``synapse.example.com:443``). There are 2 ways to do this:
```

Comment:
Maybe mention somewhere around here that both ways are described more in-depth in dedicated sections below? This section ending with "If all goes well, you should be able to connect to your server with a client and then join a room via federation" would make me think that I should have delegation working by reaching this point of the documentation.

Reply:
I reworded